src/extractcode/archive.py

```diff
@@ -103,13 +104,15 @@ def can_extract(location):
         return True
 
 
-def should_extract(location, kinds):
+def should_extract(location, kinds, ignored_extensions=()):
     """
     Return True if this location should be extracted based on the provided
     kinds
     """
     location = os.path.abspath(os.path.expanduser(location))
-    if get_extractor(location, kinds):
+    ignored_extensions = {extension : 'User ignore: Supplied by --ignore' for extension in ignored_extensions}
+    shouldIgnore = is_ignored(location, ignored_extensions)
```

Comment:
We tend to prefer snake_case vs camelCase ... here that would mean `shouldIgnore` should rather be `should_ignore`